File: app/src/main/java/com/example/s_tools/tools/retrofitcalls/VC.java
package com.example.s_tools.tools.retrofitcalls;

import android.content.Context;

import com.example.s_tools.MyWebService;
import com.example.s_tools.tools.Cvalues;
import com.example.s_tools.tools.ToastMy;
import com.google.gson.JsonObject;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static com.example.s_tools.entertainment.Fragemnts.movies.sectionheader.mymodel.MyMovies.CU_MA;

public class VC {
    public static final int V1=26;
    public static final String BLOGSPOT_COM="https://buddymy.blogspot.com/";

    public interface ApiCallback {
        void onResponse(boolean success, String updateinfo);
    }

    public static void check(Context context, ApiCallback callback) {
        MyWebService webService=MyWebService.retrofit.create(MyWebService.class);
        webService.checkVersion().enqueue(new Callback<JsonObject>() {
            @Override
            public void onResponse(Call<JsonObject> call, Response<JsonObject> response) {
                try {
                    if (response.body().get("status").getAsString().equals("ok")) {
                        int asInt=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("version").getAsJsonArray().get(0).getAsInt();
                        if (asInt != V1) {
                            MySharedPref.putVersioninfo(context, true);
                            ToastMy.successToast(context, "App Update Found! Please Restart The App To Continue", ToastMy.LENGTH_LONG);
                        } else {
                            String movie=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("movieurl").getAsJsonArray().get(0).getAsString();
                            String wall1=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("wallpaper").getAsJsonArray().get(0).getAsString();
                            String wall2=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("wallpaper").getAsJsonArray().get(1).getAsString();
                            String youtubetomp3=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("youtubetomp3").getAsJsonArray().get(0).getAsString();
                            MySharedPref.putMovieUrl(context, movie);
                            MySharedPref.putwall1(context, wall1);
                            MySharedPref.putwall2(context, wall2);
                            MySharedPref.putyoutubemp3(context, youtubetomp3);
                            callback.onResponse(true, null);
                        }
                    }
                } catch (Exception e) {
                    ToastMy.successToast(context, Cvalues.OFFLINE, ToastMy.LENGTH_LONG);
                }
            }

            @Override
            public void onFailure(Call<JsonObject> call, Throwable t) {
                if (t.getMessage().contains(CU_MA)) {
                    ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT);
                } else {
                    ToastMy.errorToast(context, t.getMessage().replace(CU_MA, ""), ToastMy.LENGTH_SHORT);
                }

            }
        });
    }

    public static void check_specific(Context context, boolean video, boolean audio, boolean update, ApiCallback callback) {
        MyWebService webService=MyWebService.retrofit.create(MyWebService.class);
        try {
            webService.checkVersion().enqueue(new Callback<JsonObject>() {
                @Override
                public void onResponse(Call<JsonObject> call, Response<JsonObject> response) {
                    try {
                        if (response.body().get("status").getAsString().equals("ok")) {
                            int asInt=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("version").getAsJsonArray().get(0).getAsInt();
                            if (asInt != V1) {
                                ToastMy.successToast(context, "App Update Available! ", ToastMy.LENGTH_SHORT);
                                MySharedPref.putVersioninfo(context, true);
                            }
                            if (audio) {
                                String youtubetomp3=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("youtubetomp3").getAsJsonArray().get(0).getAsString();
                                MySharedPref.putyoutubemp3(context, youtubetomp3);
                                callback.onResponse(true, null);
                            } else if (video) {
                                String part1=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("youtubevideo").getAsJsonArray().get(0).getAsString().split(",")[0];
                                String part2=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("youtubevideo").getAsJsonArray().get(0).getAsString().split(",")[1];
                                MySharedPref.putyoutubeVideo_part1(context, part1);
                                MySharedPref.putyoutubeVideo_part2(context, part2);
                                callback.onResponse(true, null);
                            } else if (update) {
                                String update_urls=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("newupdateurl").getAsJsonArray().get(0).getAsString();
                                callback.onResponse(true, update_urls);

                            }
                        }
                    } catch (Exception e) {
                        callback.onResponse(false, null);
                    }
                }

                @Override
                public void onFailure(Call<JsonObject> call, Throwable t) {
                    callback.onResponse(false, null);
                    if (t.getMessage().contains(CU_MA)) {
                        ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT);
                    } else {
                        ToastMy.errorToast(context, t.getMessage().replace(CU_MA, ""), ToastMy.LENGTH_SHORT);
                    }

                }
            });

        } catch (Exception e) {
            ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT);
            callback.onResponse(false, null);
        }
    }

    public static void check_v(Context context, ApiCallback callback) {
        MyWebService webService=MyWebService.retrofit.create(MyWebService.class);
        webService.checkVersion().enqueue(new Callback<JsonObject>() {
            @Override
            public void onResponse(Call<JsonObject> call, Response<JsonObject> response) {
                if (response.body().get("status").getAsString().equals("ok")) {
                    int asInt=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("version").getAsJsonArray().get(0).getAsInt();
                    if (asInt != V1) {
                        MySharedPref.putVersioninfo(context, true);
                        String update_urls=response.body().getAsJsonObject("page").getAsJsonObject("custom_fields").get("newupdateurl").getAsJsonArray().get(0).getAsString();
                        callback.onResponse(true, update_urls);
                    } else {
                        callback.onResponse(true, null);
                    }

                } else {
                    ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT);
                    callback.onResponse(false, null);
                }
            }

            @Override
            public void onFailure(Call<JsonObject> call, Throwable t) {
                callback.onResponse(false, null);
                if (t.getMessage().contains(CU_MA)) {
                    ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT);
                } else {
                    ToastMy.errorToast(context, t.getMessage().replace(CU_MA, ""), ToastMy.LENGTH_SHORT);
                }

            }
        });
    }
}

File: app/src/main/java/com/example/s_tools/tools/retrofitcalls/VC.kt
package com.example.s_tools.tools.retrofitcalls

import android.content.Context
import com.example.s_tools.tools.retrofitcalls.MySharedPref.putVersioninfo
import com.example.s_tools.tools.retrofitcalls.MySharedPref.putMovieUrl
import com.example.s_tools.tools.retrofitcalls.MySharedPref.putwall1
import com.example.s_tools.tools.retrofitcalls.MySharedPref.putwall2
import com.example.s_tools.tools.retrofitcalls.MySharedPref.putyoutubemp3
import com.example.s_tools.tools.retrofitcalls.MySharedPref.putyoutubeVideo_part1
import com.example.s_tools.tools.retrofitcalls.MySharedPref.putyoutubeVideo_part2
import com.example.s_tools.MyWebService
import com.google.gson.JsonObject
import com.example.s_tools.tools.retrofitcalls.MySharedPref
import com.example.s_tools.tools.ToastMy
import com.example.s_tools.tools.Cvalues
import com.example.s_tools.entertainment.Fragemnts.movies.sectionheader.mymodel.MyMovies
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.lang.Exception

object VC {
    const val V1 = 26
    const val BLOGSPOT_COM = "https://buddymy.blogspot.com/"
    @JvmStatic
    fun check(context: Context?, callback: (Boolean,String?)->Unit) {
        val webService = MyWebService.retrofit.create(MyWebService::class.java)
        webService.checkVersion()!!.enqueue(object : Callback<JsonObject?> {
            override fun onResponse(call: Call<JsonObject?>, response: Response<JsonObject?>) {
                try {
                    if (response.body()!!["status"].asString == "ok") {
                        val asInt = response.body()!!.getAsJsonObject("page")
                            .getAsJsonObject("custom_fields")["version"].asJsonArray[0].asInt
                        if (asInt != V1) {
                            putVersioninfo(context, true)
                            ToastMy.successToast(
                                context,
                                "App Update Found! Please Restart The App To Continue",
                                ToastMy.LENGTH_LONG
                            )
                        } else {
                            val movie = response.body()!!.getAsJsonObject("page")
                                .getAsJsonObject("custom_fields")["movieurl"].asJsonArray[0].asString
                            val wall1 = response.body()!!.getAsJsonObject("page")
                                .getAsJsonObject("custom_fields")["wallpaper"].asJsonArray[0].asString
                            val wall2 = response.body()!!.getAsJsonObject("page")
                                .getAsJsonObject("custom_fields")["wallpaper"].asJsonArray[1].asString
                            val youtubetomp3 = response.body()!!.getAsJsonObject("page")
                                .getAsJsonObject("custom_fields")["youtubetomp3"].asJsonArray[0].asString
                            putMovieUrl(context, movie)
                            putwall1(context, wall1)
                            putwall2(context, wall2)
                            putyoutubemp3(context, youtubetomp3)
                            callback(true, null)
                        }
                    }
                } catch (e: Exception) {
                    ToastMy.successToast(context, Cvalues.OFFLINE, ToastMy.LENGTH_LONG)
                }
            }

            override fun onFailure(call: Call<JsonObject?>, t: Throwable) {
                if (t.message!!.contains(MyMovies.CU_MA)) {
                    ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT)
                } else {
                    ToastMy.errorToast(
                        context,
                        t.message!!.replace(MyMovies.CU_MA, ""),
                        ToastMy.LENGTH_SHORT
                    )
                }
            }
        })
    }

    @JvmStatic
    fun check_specific(
        context: Context?,
        video: Boolean,
        audio: Boolean,
        update: Boolean,
        callback: (Boolean,String?)->Unit
    ) {
        val webService = MyWebService.retrofit.create(MyWebService::class.java)
        try {
            webService.checkVersion()!!.enqueue(object : Callback<JsonObject?> {
                override fun onResponse(call: Call<JsonObject?>, response: Response<JsonObject?>) {
                    try {
                        if (response.body()!!["status"].asString == "ok") {
                            val asInt = response.body()!!.getAsJsonObject("page")
                                .getAsJsonObject("custom_fields")["version"].asJsonArray[0].asInt
                            if (asInt != V1) {
                                ToastMy.successToast(
                                    context,
                                    "App Update Available! ",
                                    ToastMy.LENGTH_SHORT
                                )
                                putVersioninfo(context, true)
                            }
                            if (audio) {
                                val youtubetomp3 = response.body()!!.getAsJsonObject("page")
                                    .getAsJsonObject("custom_fields")["youtubetomp3"].asJsonArray[0].asString
                                putyoutubemp3(context, youtubetomp3)
                                callback(true, null)
                            } else if (video) {
                                val part1 = response.body()!!.getAsJsonObject("page")
                                    .getAsJsonObject("custom_fields")["youtubevideo"].asJsonArray[0].asString.split(
                                    ","
                                ).toTypedArray()[0]
                                val part2 = response.body()!!.getAsJsonObject("page")
                                    .getAsJsonObject("custom_fields")["youtubevideo"].asJsonArray[0].asString.split(
                                    ","
                                ).toTypedArray()[1]
                                putyoutubeVideo_part1(context, part1)
                                putyoutubeVideo_part2(context, part2)
                                callback(true, null)
                            } else if (update) {
                                val update_urls = response.body()!!.getAsJsonObject("page")
                                    .getAsJsonObject("custom_fields")["newupdateurl"].asJsonArray[0].asString
                                callback(true, update_urls)
                            }
                        }
                    } catch (e: Exception) {
                        callback(false, null)
                    }
                }

                override fun onFailure(call: Call<JsonObject?>, t: Throwable) {
                    callback(false, null)
                    if (t.message!!.contains(MyMovies.CU_MA)) {
                        ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT)
                    } else {
                        ToastMy.errorToast(
                            context,
                            t.message!!.replace(MyMovies.CU_MA, ""),
                            ToastMy.LENGTH_SHORT
                        )
                    }
                }
            })
        } catch (e: Exception) {
            ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT)
            callback(false, null)
        }
    }

    @JvmStatic
    fun check_v(context: Context?, callback: (Boolean,String?)->Unit) {
        val webService = MyWebService.retrofit.create(MyWebService::class.java)
        webService.checkVersion()!!.enqueue(object : Callback<JsonObject?> {
            override fun onResponse(call: Call<JsonObject?>, response: Response<JsonObject?>) {
                if (response.body()!!["status"].asString == "ok") {
                    val asInt = response.body()!!.getAsJsonObject("page")
                        .getAsJsonObject("custom_fields")["version"].asJsonArray[0].asInt
                    if (asInt != V1) {
                        putVersioninfo(context, true)
                        val update_urls = response.body()!!.getAsJsonObject("page")
                            .getAsJsonObject("custom_fields")["newupdateurl"].asJsonArray[0].asString
                        callback(true, update_urls)
                    } else {
                        callback(true, null)
                    }
                } else {
                    ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT)
                    callback(false, null)
                }
            }

            override fun onFailure(call: Call<JsonObject?>, t: Throwable) {
                callback(false, null)
                if (t.message!!.contains(MyMovies.CU_MA)) {
                    ToastMy.errorToast(context, "Check Connection.", ToastMy.LENGTH_SHORT)
                } else {
                    ToastMy.errorToast(
                        context,
                        t.message!!.replace(MyMovies.CU_MA, ""),
                        ToastMy.LENGTH_SHORT
                    )
                }
            }
        })
    }

    interface ApiCallback {
        fun onResponse(success: Boolean, updateinfo: String?)
    }
}